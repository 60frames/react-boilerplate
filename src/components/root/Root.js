import React, { PropTypes } from 'react';
import { Provider } from 'react-redux';
import routes from '../../routes';
import { Router } from 'react-router';

function Root({ store, history }) {
    return (
        <Provider store={store}>
            <Router history={history} routes={routes} />
        </Provider>
    );
}

Root.propTypes = {
    store: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired
};

export default Root;
