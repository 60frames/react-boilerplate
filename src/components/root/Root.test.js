/* global jest */
/* eslint-env jasmine */

jest.mock('react-redux');
jest.mock('react-router');
jest.mock('routes', () => 'routes');

import React from 'react';
import renderer from 'react-test-renderer';
import Root from 'components/root/Root';

describe('components/root/Root', () => {

    it('renders correctly', () => {
        const component = renderer.create(
            <Root
                store={{
                    store: 'store'
                }}
                history={{
                    history: 'history'
                }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
