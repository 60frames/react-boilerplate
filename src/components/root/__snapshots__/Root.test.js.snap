exports[`components/root/Root renders correctly 1`] = `
<Provider
  store={
    Object {
      "store": "store",
    }
  }>
  <Router
    history={
      Object {
        "history": "history",
      }
    }
    routes="routes" />
</Provider>
`;
